src/Silk.Core/Commands/Tests/TagCommand.cs

```diff
@@ -0,0 +1,270 @@
+
+            if (tag.OwnerId != ctx.User.Id && (!user?.Flags.Has(UserFlag.Staff) ?? true))
+            {
+                await ctx.RespondAsync("You either do not own this tag, or are not staff!");
+                return;
+            }
+
+            await _tagService.RemoveTagAsync(tagName, ctx.Guild.Id);
+            
+            string message = 
+                tag.OriginalTag is not null ?
+                "Alias successfully deleted." :
+                tag.Aliases?.Any() ?? false ?
+                    "Tag and all corresponded aliases successfully deleted." :
+                    "Tag successfully deleted.";
+            
+            await ctx.RespondAsync(message);
+
+        }
+
+        [Command]
+        public async Task Edit(CommandContext ctx, string tagName, [RemainingText] string content)
+        {
+            var couldEditTag = await _tagService.UpdateTagContentAsync(tagName, content, ctx.User.Id, ctx.User.Id);
+            if (!couldEditTag.success)
+            {
+                await ctx.RespondAsync(couldEditTag.reason);
+            }
+            else
+            {
+                await ctx.RespondAsync("Sucessfully edited tag!");
+            }
+        }
+        
+        [Command]
+        public async Task Search(CommandContext ctx, string tagName)
+        {
+            var tags = await _mediator.Send(new TagRequest.GetByName(tagName, ctx.Guild.Id));
+            if (tags is null)
+            {
+                await ctx.RespondAsync("No tags found :c");
+                return;
+            }
+            
+            string allTags = string.Join("\n\n", tags!
+                .Select(t =>
+                {
+                    var s = $"`{t.Name}`";
+                    if (t.OriginalTagId is not null)
+                    {
+                        s += $" → `{t.OriginalTag!.Name}`";
+                    }
+                    s += $" - <@{t.OwnerId}>";
+                    return s;
+                }));
+            var builder = new DiscordEmbedBuilder()
+                .WithColor(DiscordColor.Blurple)
+                .WithTitle($"Result for {tagName}:")
+                .WithFooter($"Silk! | Requested by {ctx.User.Id}");
+            
+            if (tags.Count() < 10)
+            {
+                builder.WithDescription(allTags);
+                await ctx.RespondAsync(builder);
+            }
+            else
+            {
+                var interactivity = ctx.Client.GetInteractivity();
+
+                var pages = interactivity.GeneratePagesInEmbed(allTags, SplitType.Line, builder);
+                await interactivity.SendPaginatedMessageAsync(ctx.Channel, ctx.User, pages);
+            }
+            
+        }
+        
+    }
+    
+    [RequireGuild]
+    [Category(Categories.Server)]
+    public class TagsCommand : BaseCommandModule
+    {
+        private readonly TagService _tagService;
+        public TagsCommand(TagService tagService)
+        {
```

Comment:
What's a lambda?

Reply:
It's done for consistency. I could use an inline constructor, but it'd need to refactored should this class ever require more dependancies.